koku/masu/api/upgrade_trino/util/verify_parquet_files.py

```diff
@@ -0,0 +1,302 @@
+import logging
+import os
+import uuid
+from pathlib import Path
+
+import ciso8601
+import pandas as pd
+import pyarrow as pa
+import pyarrow.parquet as pq
+from botocore.exceptions import ClientError
+from django.conf import settings
+from django_tenants.utils import schema_context
+
+from api.common import log_json
+from api.provider.models import Provider
+from masu.api.upgrade_trino.util.state_tracker import StateTracker
+from masu.config import Config
+from masu.processor.parquet.parquet_report_processor import OPENSHIFT_REPORT_TYPE
+from masu.util.aws.common import get_s3_resource
+from masu.util.common import get_path_prefix
+
+
+LOG = logging.getLogger(__name__)
+
+
+class VerifyParquetFiles:
+    CONVERTER_VERSION = 1.0
+    S3_OBJ_LOG_KEY = "s3_object_key"
+    S3_PREFIX_LOG_KEY = "s3_prefix"
+
+    def __init__(self, schema_name, provider_uuid, provider_type, simulate, bill_date, cleaned_column_mapping):
+        self.schema_name = schema_name
+        self.provider_uuid = uuid.UUID(provider_uuid)
+        self.provider_type = provider_type.replace("-local", "")
+        self.simulate = simulate
+        self.bill_date = bill_date
+        self.file_tracker = StateTracker(provider_uuid)
+        self.report_types = self._set_report_types()
+        self.required_columns = self._set_pyarrow_types(cleaned_column_mapping)
+        self.logging_context = {
+            "provider_type": self.provider_type,
+            "provider_uuid": self.provider_uuid,
+            "schema": self.schema_name,
+            "simulate": self.simulate,
+            "bill_date": self.bill_date,
+        }
+
+    def _set_pyarrow_types(self, cleaned_column_mapping):
+        mapping = {}
+        for key, default_val in cleaned_column_mapping.items():
+            if pd.isnull(default_val):
+                # TODO: Azure saves datetime as pa.timestamp("ms")
+                # TODO: AWS saves datetime as timestamp[ms, tz=UTC]
```

Comment:
Also I believe OCI is doing `ms, tz=UTC` too, so depending the route we might need to change that too.

Reply:
We can add a check to switch this for azure then. 👍